### ISSUE TITLE
Unfix of SSRF in webhook 

### ISSUE THREAD

**Issue by cokeBeer** [gogs/gogs#7031]
### Gogs version

v0.12.9

### Git version

- Server:  2.24.1
- Client:  2.24.1

### Operating system

MacOS 10.15.4 

### Database

SQLite3

### Describe the bug

in changelog of 0.12.9,  https://github.com/gogs/gogs/issues/6901 are claimed to be fixed. But still exist via test.

### To reproduce

see https://github.com/cokeBeer/go-cves/blob/main/CVE-2022-1285/CVE-2022-1285.md

### Expected behavior

request to internal ip blocked

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by unknwon** [comment]
Thanks for the feedback!

Please see https://github.com/gogs/gogs/issues/5372#issuecomment-1059938519

**Comment by unknwon** [comment]
Note: disable follow redirects in HTTP client.

**Comment by Sim4n6** [comment]
Indeed, this is what lead to this security report https://huntr.com/bounties/f57e0882-8a85-4120-97a3-de7bdb8965a4 ...

**Comment by Sim4n6** [comment]
Hi, any update on this please?

**Comment by unknwon** [comment]
Hi @Sim4n6, based on the linked report, the current status is "Awaiting fix". If you feel about it, PRs welcome :D

**Comment by Sim4n6** [comment]
Do you have any bug bounty reward program for identifying vulnerabilities and suggesting fixes please?
